src/Plugin.php

```diff
@@ -48,6 +48,7 @@ public static function plugins_loaded() {
     add_filter('network_site_url', __NAMESPACE__ . '\Security::site_url', 100, 3);
     add_filter('site_url', __NAMESPACE__ . '\Security::site_url', 100, 3);
     add_filter('wp_redirect', __NAMESPACE__ . '\Security::wp_redirect', 1);
+    add_filter('wp_headers', __NAMESPACE__ . '\Security::additional_securityheaders', 1);
```

Comment:
1. The Content Security Policy is a different kind of security layer then existing stuff here (which are protections against script kiddies).  Can you move this into a new ContentSecurity class?

2. The callback method name should match the hook name, or it should otherwise clarify what is doing (adding response headers).

3. Why did you use the priority 1?  Whenever using a priority different to the default, there should be an inline comment that explains before or after which other thing this registered hook is supposed to run.  Other code can change their priorities, at which point it becomes hard to debug what the originally intended integration was.  Given that there is no comment or commit stating something in relation to it, I assume the priority 1 was a premature optimization, so I would recommend to remove it.

```suggestion
    add_filter('wp_headers', __NAMESPACE__ . '\ContentSecurity::wp_headers');
```

Reply:
In this case I've changed the original approach I took and followed your suggestion to use the htaccess rules instead, so I moved all the logic from here to the existing `Schema` class which is the one in charge of this job.